ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -0,0 +1,330 @@
+      -- non-deterministic.
+    , getIsFetched       :: STM m (Point blk -> Bool)
+
+      -- | Stream blocks
+      --
+      -- Streaming is not restricted to the current fork, but there must be
+      -- an unbroken path from the starting point to the end point /at the time
+      -- of initialization/ of the iterator. Once the iterator has been
+      -- initialized, it will not be affected by subsequent calls to 'addBlock'.
+      --
+      -- The iterator /does/ have a limited lifetime, however. The chain DB
+      -- internally partitions the chain into an " immutable " part and a
+      -- " volatile " part, moving blocks from the volatile DB to the immutable
+      -- DB when they become more than @k@ deep into the chain. When a block
+      -- with slot number @n@ is added to the immutble DB, a time delay @t@
+      -- kicks in; after that time delay expires, all blocks older than @n@ may
+      -- be removed from the volatile DB, /including any blocks that happen to
+      -- live on other forks/ (since those forks must now, by definition, be too
+      -- distant). This time delay @t@ also provides a worst-case bound for the
+      -- lifetime of the iterator: if the iterator traverses a chain that
+      -- forks off from our current chain at the tip of the immutable DB,
+      -- then the first block on that fork will become unavailable as soon as
+      -- another block is pushed to the current chain and the subsequent
+      -- time delay expires.
+    , streamBlocks       :: StreamFrom blk -> StreamTo blk -> m (Iterator m blk)
+
+      -- | Chain reader
+      --
+      -- A chain reader is an iterator that tracks the state of the /current/
+      -- chain: calling @next@ on the iterator will either give you the next
+      -- block, or (if we have switched to a fork) the instruction to rollback.
+      --
+      -- The tracking iterator starts at genesis (see also 'trackForward').
+    , newReader          :: m (Reader m hdr)
+
+      -- | Known to be invalid blocks
+    , knownInvalidBlocks :: STM m (Set (Point blk))
+
+      -- | Check if the specified point is on the current chain
+      --
+      -- This lives in @m@, not @STM m@, because if the point is not on the
+      -- current chain fragment, it might have to query the immutable DB.
+    , pointOnChain       :: Point blk -> m Bool
+    }
+
+{-------------------------------------------------------------------------------
+  Derived
+-------------------------------------------------------------------------------}
+
+getTipHeader :: MonadSTM m => ChainDB m blk hdr -> STM m hdr
+getTipHeader ChainDB{..} =
+    mustExist . Fragment.head <$> getCurrentChain
+  where
+    mustExist :: Maybe a -> a
+    mustExist (Just a) = a
+    mustExist Nothing  = error "invariant violation: empty current chain"
+
+getTipPoint :: MonadSTM m => ChainDB m blk hdr -> STM m (Point blk)
+getTipPoint chainDB@ChainDB{..} =
+    Block.castPoint . Chain.blockPoint <$> getTipHeader chainDB
```

Comment:
Of course, returning `genesisPoint` really isn't correct here; but like you say, the anchored fragment work will fix that. For now I'll move this into the record also, since in the model we can implement this easily.